### PR TITLE
Test errors on yarn build 2

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
CI: Exit on error when package build fails

We were not exiting with an error when a package build was failing, we
were missing an exit command ;)

Then, the trap command was a mess, it was not properly espaced with
backslashes and was adding more complexity than anything else. I just
realized, that if we have this:

bash -c "trap cleanup EXIT;CMD1;CMD2;cleanup"

If CMD* are bash scripts, the trap call should not be needed, because the last
cleanup should always be executed, no matter how CMD2 exits.

Then, using a pipe was messing up the return code (exit code) of the
background jobs, plus the output was confusing. Meaning, we better
change this

docker run ..... | tee output.log &

by

docker run ..... > pid.log
wait pid
cat pid.log

Finally, to make sure we are not hidding an error when not running in
parallel, let's set the pipefail option, so this command

docker run .... | tee output.log

will fail if the docker run commnad failed

 * [X] No changelog needed

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
@juliogonzalez Please review https://github.com/uyuni-project/uyuni/pull/4197/commits/e209dec1f012d4f3f04e19c13adaeae0b2177445 .

Note this PR has 2 additional commits for testing this change. Basically I am making the spacewalk-web package fail to build, this is why the WIP in the title. This is for testing and you can see the result in https://ci.suse.de/blue/organizations/jenkins/uyuni-prs-ci-tests/detail/uyuni-prs-ci-tests/519/pipeline

See that the job is failing ... this is what we are testing here, that the job fails when the build fails. Obviously, only https://github.com/uyuni-project/uyuni/pull/4197/commits/e209dec1f012d4f3f04e19c13adaeae0b2177445 should be part of the final PR.

File: susemanager-utils/testing/automation/push-to-obs.sh [hunk 1]
```diff
@@ -73,20 +74,23 @@ echo "Starting building and submission at $(date)"
...
     CMD="/manager/susemanager-utils/testing/docker/scripts/push-to-obs.sh -d '${DESTINATIONS}' -c /tmp/.oscrc -p '${p}' ${VERBOSE} ${TEST} ${OBS_TEST_PROJECT} ${EXTRA_OPTS}"
     # Set pipefail so the result of a pipe is different to zero if one of the commands fail.
     # This way, we can add tee at the end but be sure the result would be non zero if the command before has failed.
-    set -o pipefail
     if [ "$PARALLEL_BUILD" == "TRUE" ];then
-        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "trap \"EXIT_CODE=\$?;${CHOWN_CMD};exit \$EXITCODE \" EXIT;${INITIAL_CMD}; ${CMD}; RET=\${?}; ${CHOWN_CMD} && exit \${RET}" | tee ${GITROOT}/logs/${p}.log &
-        PIDS="$PIDS $!"
+        echo "Building ${p} in parallel"
+        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "${INITIAL_CMD};${CMD};RET=\${?};${CHOWN_CMD} && exit \${RET}" 2>&1 > ${GITROOT}/logs/${p}.log &
+        pid=$!
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/push-to-obs.sh:87]:
```suggestion
        pid=${!}
```

File: susemanager-utils/testing/automation/push-to-obs.sh [hunk 2]
```diff
@@ -73,20 +74,23 @@ echo "Starting building and submission at $(date)"
...
     CMD="/manager/susemanager-utils/testing/docker/scripts/push-to-obs.sh -d '${DESTINATIONS}' -c /tmp/.oscrc -p '${p}' ${VERBOSE} ${TEST} ${OBS_TEST_PROJECT} ${EXTRA_OPTS}"
     # Set pipefail so the result of a pipe is different to zero if one of the commands fail.
     # This way, we can add tee at the end but be sure the result would be non zero if the command before has failed.
-    set -o pipefail
     if [ "$PARALLEL_BUILD" == "TRUE" ];then
-        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "trap \"EXIT_CODE=\$?;${CHOWN_CMD};exit \$EXITCODE \" EXIT;${INITIAL_CMD}; ${CMD}; RET=\${?}; ${CHOWN_CMD} && exit \${RET}" | tee ${GITROOT}/logs/${p}.log &
-        PIDS="$PIDS $!"
+        echo "Building ${p} in parallel"
+        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "${INITIAL_CMD};${CMD};RET=\${?};${CHOWN_CMD} && exit \${RET}" 2>&1 > ${GITROOT}/logs/${p}.log &
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/push-to-obs.sh:86]:
```suggestion
        docker run --rm=true -v ${GITROOT}:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc ${REGISTRY}/${PUSH2OBS_CONTAINER} /bin/bash -c "${INITIAL_CMD};${CMD};RET=\${?};${CHOWN_CMD} && exit \${RET}" 2>&1 > ${GITROOT}/logs/${p}.log &
```

File: susemanager-utils/testing/automation/push-to-obs.sh [hunk 3]
```diff
@@ -73,20 +74,23 @@ echo "Starting building and submission at $(date)"
...
+        pid=$!
+        PIDS="${PIDS} ${pid}"
+        ln -s ${GITROOT}/logs/${p}.log ${GITROOT}/logs/${pid}.log
     else
-        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "trap \"EXIT_CODE=\$?;${CHOWN_CMD};exit \$EXITCODE\" EXIT;${INITIAL_CMD}; ${CMD}; RET=\${?}; ${CHOWN_CMD} &&  exit \${RET}" | tee ${GITROOT}/logs/${p}.log
+        echo "Building ${p}"
+        docker run --rm=true -v $GITROOT:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc $REGISTRY/$PUSH2OBS_CONTAINER /bin/bash -c "${INITIAL_CMD};${CMD};RET=\${?};${CHOWN_CMD} && exit \${RET}" | tee ${GITROOT}/logs/${p}.log
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/push-to-obs.sh:92]:
```suggestion
        docker run --rm=true -v ${GITROOT}:/manager -v /srv/mirror:/srv/mirror --mount type=bind,source=${CREDENTIALS},target=/tmp/.oscrc ${REGISTRY}/${PUSH2OBS_CONTAINER} /bin/bash -c "${INITIAL_CMD};${CMD};RET=\${?};${CHOWN_CMD} && exit \${RET}" | tee ${GITROOT}/logs/${p}.log
```

File: susemanager-utils/testing/automation/push-to-obs.sh [hunk 4]
```diff
@@ -99,14 +103,17 @@ for i in $PIDS;do
...
     result=$?
     echo "$i finished, result is $result"
+    cat ${GITROOT}/logs/${i}.log
     if [ $result -ne 0 ];then
         echo "Seems there was an error with pid $i"
         PRET=-1
     fi
 done
 
+
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/push-to-obs.sh:113]: Do we need this empty line?

File: susemanager-utils/testing/automation/push-to-obs.sh [hunk 5]
```diff
@@ -99,14 +103,17 @@ for i in $PIDS;do
...
 if [ $PRET -ne 0 ];then
     echo "Review the logs."
+    exit $PRET
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/push-to-obs.sh:116]:
```suggestion
    exit ${PRET}
```

**Review by juliogonzalez — APPROVED** [review]
Left comments for some style issues, but other than that the PR looks good and makes sense :-)

I guess you will need to backport it.

File: susemanager-utils/testing/automation/push-to-obs.sh (hunk 4 above)
Comment (jordimassaguerpla) [reply susemanager-utils/testing/automation/push-to-obs.sh:113]: Not really ;)

**Comment by jordimassaguerpla** [comment]
Thanks @juliogonzalez for your revision. I have applied your suggestions and removed the test commits. We can merge this now.

**Pull request merged**